### PR TITLE
fix: update user overview in use rmanagement

### PR DESCRIPTION
**Pull request by harjibbolar26**
# Description

This PR Fixes count of the total user, deleted user and active user in the user management dashboard of the superadmin in the route /dashboard/admin/user. It also adds the modal that shows that a user was successfully deleted.

**Closes #issue_number_here**

# Changes proposed

## What were you told to do?

- Update the code to ensure all counts are updated everytime an action is performed. 
- Include the modal that shows a user was successfully deleted.

## What did you do?

- Synced the code to ensure total users, active user and deleted user counts are updated everytime an action is performed.
- Added a popup modal to show the user was deleted successfully.

# Check List (Check all the applicable boxes)

🚨Please review the [contribution guideline](CONTRIBUTING.md) for this repository.

- [x] My code follows the code style of this project.
- [x] This PR does not contain plagiarized content.
- [x] The title and description of the PR is clear and explains the approach.
- [x] I am making a pull request against the **dev branch** (left side).
- [x] My commit messages styles matches our requested structure.
- [x] My code additions will fail neither code linting checks nor unit test.
- [x] I am only making changes to files I was requested to.

# Screenshots/Videos

Screenshot (244)

### CODE REVIEW

File: src/app/dashboard/(admin)/admin/(overview)/users/component/dialogue/delete-successful.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import { Button } from "~/components/common/common-button";
+import {
+  Dialog,
+  DialogContent,
+  DialogDescription,
+  DialogHeader,
+  DialogTitle,
+} from "~/components/ui/dialog";
+
+const DeleteSuccessfulDialog = ({ onClose }: { onClose: () => void }) => {
```
Comment (billmal071) [inline src/app/dashboard/(admin)/admin/(overview)/users/component/dialogue/delete-successful.tsx:10]: Create a type for this outside the function.

File: src/app/dashboard/(admin)/admin/(overview)/users/page.tsx [hunk 2]
```diff
@@ -104,6 +105,8 @@ const UserPage = () => {
     FilterDataProperties | undefined
   >();
 
+  const deletedUser = 0;
```
Comment (billmal071) [inline src/app/dashboard/(admin)/admin/(overview)/users/page.tsx:108]: What's this for?

**Review by billmal071 — CHANGES_REQUESTED** [review]
Make changes.

**Review by phoenixdahdev — APPROVED** [review]

**Review by billmal071 — APPROVED** [review]
Make changes in new PR.

**Pull request merged**